cmd/namespace-lock.go

```diff
@@ -56,39 +60,45 @@ type RWLockerSync interface {
 // Returns lock clients and the node index for the current server.
 func newDsyncNodes(endpoints EndpointList) (clnts []dsync.NetLocker, myNode int) {
 	cred := globalServerConfig.GetCredential()
-	clnts = make([]dsync.NetLocker, len(endpoints))
 	myNode = -1
-	for index, endpoint := range endpoints {
+	seenHosts := set.NewStringSet()
+	i := 0
+	for _, endpoint := range endpoints {
+		if seenHosts.Contains(endpoint.Host) {
+			continue
+		}
+		seenHosts.Add(endpoint.Host)
 		if !endpoint.IsLocal {
 			// For a remote endpoints setup a lock RPC client.
-			clnts[index] = newLockRPCClient(authConfig{
+			clnts = append(clnts, newLockRPCClient(authConfig{
 				accessKey:       cred.AccessKey,
 				secretKey:       cred.SecretKey,
 				serverAddr:      endpoint.Host,
 				secureConn:      globalIsSSL,
-				serviceEndpoint: pathutil.Join(minioReservedBucketPath, lockServicePath, endpoint.Path),
+				serviceEndpoint: pathutil.Join(minioReservedBucketPath, lockServicePath),
 				serviceName:     lockServiceName,
-			})
+			}))
+			i++
 			continue
 		}
 
 		// Local endpoint
 		if myNode == -1 {
-			myNode = index
+			myNode = i
```

Comment:
Can you not say: `len(clnts)` -- and remove `'i` altogether ?

Reply:
Oh yes definitely. @fwessels